### PR TITLE
[libdatachannel] update to 0.18.2

### LINKED ISSUE #29527: [libdatachannel] update to <0.18.2>
Library name: libdatachannel

New version number: 0.18.2

Other information that may be useful (release notes, etc...)

https://github.com/paullouisageneau/libdatachannel/releases

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/29527

Note: Update `libdatachannel` version to 0.18.2
All features passed with following triplets:
```
x86-windows
x64-windows
x64-windows-static
```
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libdatachannel/portfile.cmake [hunk 1]
```diff
@@ -47,4 +47,5 @@ ${DATACHANNEL_CONFIG}")
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include" "${CURRENT_PACKAGES_DIR}/debug/share")
 
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+#file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (FrankXie05) [inline ports/libdatachannel/portfile.cmake:50]:
```suggestion
```

File: ports/libdatachannel/vcpkg.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "libdatachannel",
-  "version-semver": "0.17.11",
+  "version-semver": "0.18.2",
   "description": "libdatachannel is a standalone implementation of WebRTC Data Channels, WebRTC Media Transport, and WebSockets in C++17 with C bindings for POSIX platforms (including GNU/Linux, Android, and Apple macOS) and Microsoft Windows.",
   "homepage": "https://github.com/paullouisageneau/libdatachannel",
-  "license": "LGPL-2.1-only",
+  "license": "MPL-2.0",
```
Comment (FrankXie05) [inline ports/libdatachannel/vcpkg.json:6]:
```suggestion
  "license": "MPL-2.0",
```
Why amend the license？

File: ports/libdatachannel/portfile.cmake [hunk 3]
```diff
@@ -7,8 +7,8 @@ endif()
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO paullouisageneau/libdatachannel
-    REF 2c6ef2053c6d28a66c16d0dae54e432750753575 #v0.17.11
-    SHA512 b65cff5a7cb274fe4651c2cd395533c50eb2959192596e6ac7017af40c93d75cbc83ad86f8810e728b33d8d0b5993eddba4eac8644a603c5e641bf7cc4b87425
+    REF "v${VERSION}" #v0.17.11
```
Comment (FrankXie05) [inline ports/libdatachannel/portfile.cmake:10]:
```suggestion
    REF "v${VERSION}" #v0.18.2

```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/libdatachannel/vcpkg.json (hunk 2 above)
Comment (jimwang118) [reply ports/libdatachannel/vcpkg.json:6]: The license has been updated upstream, please see for details.https://github.com/paullouisageneau/libdatachannel/commit/a0f3f01bf5fdf9336bef830879e5bec07a8bca73

**Review by LilyWangLL — APPROVED** [review]

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**